dockerfiles/init/manifests/codenvy.env

```diff
@@ -86,17 +86,121 @@
+#           and exposed ports
+#        2. If address is missing, if server connects over Unix socket, then use 
+#           localhost and exposed ports
+#        3. Else, use DOCKER_HOST and published ports
+#
+# Browser --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
+#        1. If set use the value of CODENVY_DOCKER_IP_EXTERNAL
+#        2. Else if set use the value of CODENVY_DOCKER_IP
+#        3. Else, if server connects over Unix socket, then use localhost
+#        4. Else, use DOCKER_HOST
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'docker-local':
+#        1. If set use the value of CODENVY_DOCKER_IP_EXTERNAL
+#        2. Else use the address of the workspace container within the docker network, 
+#           if it is set
+#        3. If address is missing, if server connects over Unix socket, then use 
+#           localhost
+#        4. Else, use DOCKER_HOST
+#
+# Workspace Agent --> Codenvy Server
+#    1. Default is 'http://codenvy-host:${SERVER_PORT}/wsmaster/api', where 
+#       'codenvy-host' is IP of server.
+#    2. Else, use value of CODENVY_WORKSPACE_CHE__SERVER__ENDPOINT
+#    3. Else, if 'docker0' interface is unreachable, then 'codenvy-host' replaced 
+#       with 172.17.42.1 or 192.168.99.1
+#    4. Else, print connection exception
+
+# Codenvy Server API Endpoint
+#     The location of the API end point where dashboard and IDE clients will look for
+#     interacting with the Codenvy server, which we also call the workspace master.
+#CODENVY_API=http://localhost:${SERVER_PORT}/wsmaster/api
+
+# Docker Daemon URL
+#     How the Codenvy server will discover the location of the Docker daemon. If this is
+#     not set, then Codenvy will use DOCKER_HOST or unix:///var/run/docker.sock.
+#     It would be rare to need to set this as most clients set DOCKER_HOST or volume
+#     mount a Docker socket when starting the Codenvy CLI.
+#CODENVY_DOCKER_DAEMON__URL=NULL
+
+# Docker Host
+#     How Codenvy will connect to the Docker host if CODENVY_DOCKER_DAEMON__URL is not 
+#     set. This value can be provided on the command line as well. If this value is
+#     not set, then Codenvy will use the volume mount of unix:///var/run/docker.sock.
+#DOCKER_HOST=tcp://localhost:2375
+
+# Workspace Address Resolution Strategy
+#     The strategy to determine how workspace servers users launch will be exposed. 
+#     If the server and your workspaces are on different networks that flow through a
+#     firewall, those firewalls can prevent communications between Codenvy and workspaces.
+#     The default strategy exposes workspaces with the IP of Docker and ephemeral ports.
+#     A Docker local strategy uses internal Docker network IP addresses and exposed ports.
+#
+#     Options:
+#       - 'default':      internal address is DOCKER_HOST with ephemeral ports
+#       - 'docker-local': internal address is Docker network container address with
```

Comment:
Codenvy uses another approach. It can depends on url transformer property but by default we proxy all request using our HAProxy container.